encodings/fastlanes/src/bitpacking/compress.rs

```diff
@@ -96,22 +98,27 @@ pub fn bitpack_patches(
     parray: &PrimitiveArray,
     bit_width: usize,
     num_exceptions_hint: usize,
-) -> Array {
+) -> Option<Array> {
     match_each_integer_ptype!(parray.ptype(), |$T| {
         let mut indices: Vec<u64> = Vec::with_capacity(num_exceptions_hint);
         let mut values: Vec<$T> = Vec::with_capacity(num_exceptions_hint);
         for (i, v) in parray.maybe_null_slice::<$T>().iter().enumerate() {
-            if (v.leading_zeros() as usize) < parray.ptype().bit_width() - bit_width {
+            if (v.leading_zeros() as usize) < parray.ptype().bit_width() - bit_width && parray.is_valid(i) {
                 indices.push(i as u64);
                 values.push(*v);
             }
         }
-        SparseArray::try_new(
-            indices.into_array(),
-            PrimitiveArray::from_vec(values, Validity::AllValid).into_array(),
-            parray.len(),
-            Scalar::null(parray.dtype().as_nullable()),
-        ).unwrap().into_array()
+
+        (!indices.is_empty()).then(|| {
+            SparseArray::try_new(
+                indices.into_array(),
+                PrimitiveArray::from_vec(values, Validity::AllValid).into_array(),
+                parray.len(),
+                Scalar::null(parray.dtype().as_nullable()),
+            )
+            .unwrap()
```

Comment:
no unwrapping! (how was this not caught by clippy?)

Reply:
hahaha, no clue. It was like this before fwiw

Reply:
ok, fixed